batcher_test.go

```
@@ -50,105 +51,85 @@ func TestPGXBatcher_Execute_Errors(t *testing.T) {
 			t.Errorf("Expected 1 error, but got %d", len(errs))
 		}
 
-		if errs[0].sql != "INVALID SQL" {
+		/*if errs[0].sql != "INVALID SQL" {
 			t.Errorf("Expected error SQL to be 'INVALID SQL', but got '%s'", errs[0].sql)
-		}
+		}*/
 	} else {
 		t.Errorf("Expected error of type 'BatcherErrors', but got '%T'", err)
 	}
 }
 
 func TestPGXBatcher_Execute_Transactional_Errors(t *testing.T) {
-
-	batcher := New(conn, true)
+	b := New(conn, true)
 
 	// add invalid SQL statement to batch
-	batcher.Queue("INVALID SQL", []interface{}{})
+	b.Queue("INVALID SQL")
 
 	// execute batch
-	err := batcher.Execute(context.TODO())
+	err := b.Execute(context.TODO())
 
 	// assert error type and message
 	if err == nil {
 		t.Error("Expected error, but got nil")
 	}
 
 	if errs, ok := err.(StatementErrors); ok {
-		if len(errs) != 1 {
-			t.Errorf("Expected 1 error, but got %d", len(errs))
+		if len(errs) != 3 {
```

You may want to revisit this, there are three errors because there are items still in the queue, meaning this was not a new initialisation at the start of the test

---

3 errors - because there are 3 requests (BEGIN, INSERT, COMMIT). If you look at batcher.go, line 99-104, you will understand why there are 3 errors.

---

https://github.com/jackc/pgx/blob/master/batch.go#L285

---

I see, I'm not sure how rewriting this is better for the user? The issue is with the query, not the transactional statements?

---

Maybe return the first error? Because batch is an implicit transaction of all requests.
https://github.com/jackc/pgx/issues/1066